apps/call.go

```diff
@@ -133,6 +133,25 @@ type CallResponse struct {
 	Form *Form `json:"form,omitempty"`
 }
 
+// ProxyCallResponse contains everything the CallResponse struct contains, plus some additional
+// data for the client, such as information about the App's bot account.
+//
+// Apps will use the CallResponse struct to respond to a CallRequest, and the proxy will
+// decorate the response using the ProxyCallResponse to provide additional information.
+type ProxyCallResponse struct {
+	*CallResponse
+
+	// Used to provide info about the App to client, e.g. the bot user id
+	AppMetadata *AppMetadataForClient `json:"app_metadata"`
+}
```

Comment:
@levb We're introducing more pointers here. Should we instead always use a struct for `CallResponse` throughout the codebase instead of a pointer? The client expects `AppMetadata` to be optional, since it makes its own transient `CallResponse`s to create local errors, so that should actually be a pointer.

Reply:
2/5 Just embed the fields themselves, no need for a new struct?

Reply:
@levb The PR is already merged on mobile that uses this struct https://github.com/mattermost/mattermost-mobile/blob/06be22d6f72af8aad42af3dd3a0ef18bd54108c0/app/mm-redux/types/apps.ts#L76-L91

Reply:
Ouch... OK. Can we file a ticket to rename this all later, since it doesn't affect dev experience?

Reply:
@levb If we wanted to change it, we should change it now before any of the clients are released. Wouldn't be a PITA to change it after the first release?

Reply:
@larkox What are your thoughts on this?

Reply:
I like having a separated field, in case we need to add more stuff in the future. I don't want to pollute much more the response (we already have at top level markdown, error, navigate to url, use external browser... I like the app_metadata at least being in its own compartment). But 0/5.

Reply:
@larkox Hmm, we control that namespace (`App`), so as long as we don't create conflicts, why do we need the extra level? I prefer that we consider that every "exported" type (even for internal use) comes with a "complexity cost". (code has a much lower complexity cost since it's somewhat self-explanatory, and contained).